### DISCUSSION TITLE
A Question on Why Padding is Used with Masking in MRI Data Undersampling

### DISCUSSION THREAD

**Discussion by cuiyushuai02** [facebookresearch/fastMRI#352] [Q&A] [answered]
Hello, I am very interested in your work and have a question regarding data undersampling that I'd like to consult with you on:

Using the information provided in the h5 file, we compute `attrs["padding_left"]` and `attrs["padding_right"]` in the `_retrieve_metadata` function, and then use these two attributes in the `apply_mask` method to pad certain parts of the image with zeros on the left and right.

I don’t understand how the padding attributes are calculated in the `_retrieve_metadata` function, nor do I understand why padding with zeros is applied to the image along with the use of the mask. Could you provide some resources to help me learn about this part?

Thank you very much for sharing this code.

**Comment by mmuckley** [comment, marked as answer]
Hello @cuiyushuai02, this is related to understanding the Siemens reconstruction process. When a technician specifies the scan parameters, including resolution and matrix size, sometimes to achieve an isotropic resolution the algorithm will pad a few zeros on an axis prior to inverse Fourier transform. So it's really upstream of the fastMRI data.

For more details I'd refer to Siemens documentation and standard MRI references. I don't know of any open-source variants - most knowledge of this comes from textbooks and PDF documents shared among researchers in MRI labs.
